### PR TITLE
Bug 1257767 - [TV][2.5] Showing app icon under preview window splash screen's spinner

### PR DESCRIPTION
**Pull request by josyeh**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=20fab49e32814ae5cf25b5bbd04dee886961099c

File: tv_apps/smart-system/js/splash_screen.js [hunk 1]
```diff
@@ -56,5 +63,45 @@
...
+    var url = icons[maxIconSize];
+
+    if (!url) {
+      return;
+    }
+
+    if (url.indexOf('data:') === 0 ||
+        url.indexOf('app://') === 0 ||
+        url.indexOf('http://') === 0 ||
+        url.indexOf('https://') === 0) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/splash_screen.js]: I would use `startsWith()` [1] or `match(regexp)` [2] to simplify these conditions, but it's up to you.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
[2] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

File: tv_apps/smart-system/js/splash_screen.js [hunk 2]
```diff
@@ -56,5 +63,45 @@
...
+    } else if (app.features.iconUrl) {
+      return app.features.iconUrl;
+    }
+  };
+
+  SplashScreen.prototype.getIconFromManifestURL = function(manifestURL) {
+    var app = applications.getByManifestURL(manifestURL);
+    var icons = app.manifest.icons;
+
+    var maxIconSize = Object.keys(icons).reduce(function(prev, current) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/splash_screen.js]: There's a trick to get the largest number from an array: `Math.max(... Object.keys(icons))`, which is equal to `Math.max.apply(null, Object.keys(icons))`. :p

Again, it's up to you.

File: tv_apps/smart-system/js/splash_screen.js (hunk 1 above)
Comment (jostw) [inline tv_apps/smart-system/js/splash_screen.js]: Actually I copied it from [here] (https://github.com/mozilla-b2g/gaia/blob/master/shared/js/smart-screen/applications.js#L143). But `startsWith` seems better. Thanks.

File: tv_apps/smart-system/js/splash_screen.js (hunk 2 above)
Comment (jostw) [inline tv_apps/smart-system/js/splash_screen.js]: Cool. I love this trick :D

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fbac665d1f263489ad9159cb8e576490dc9355b3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f2ebc546d274a2c0b1f6b542ee849a529b10a8a4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3f96f2691849cafa89bee7203a10d3ae90178f9d

**Pull request merged**